ujt/callbacks/view_callbacks.py

```diff
@@ -101,38 +31,91 @@ def delete_view(delete_timestamps):
             },
             "value",
         ),
+        Input(
+            {
+                id_constants.CREATE_VIEW_BUTTON: ALL,
+            },
+            "n_clicks_timestamp",
+        ),
+        Input(
+            {
+                id_constants.DELETE_VIEW_BUTTON: id_constants.DELETE_VIEW_BUTTON,
+                "index": ALL,
+            },
+            "n_clicks_timestamp",
+        ),
     ],
-    prevent_initial_call=True,
 )
-def modify_view(tag_dropdown_values, style_dropdown_values):
-    """Updates the corresponding applied tag in the tag map.
+def update_view_store(
+    tag_dropdown_values, style_dropdown_values, create_timestamps, delete_timestamps
+) -> Tuple[List[Tuple[str, str]], str]:
+    """Saves the current list of views to the VIEW_STORE .
+
+    Notice that the style of this callback is different from the apply tag or create tag callbacks,
+    even though they both handle the same type of UI elements (dynamically creating rows).
+    This callback updates the VIEW_STORE by reading the current state of the rows of dropdown menus.
+    In turn, the generate_view_panel callback updates the dropdown menus based on the contents of the VIEW_STORE. 
+
+    Views are written to the VIEW_STORE in order to make them local for each user.
+    Thus, we must combine all the functionality into a single callback to write to the component.
+    This is in contrast to styles, tags, and applied tags, callbacks, which write
+    their respective data structure to the Dash server via the state module as a side effect.
+
+    We could theoretically refactor the tag creation panel and tag application panel to be in this style.
```

Comment:
I think this is probably too much detail to leave in code comments. A comment on the CL (so that anyone who wants to know what thoughts went into adding this here can look it up) or just an entry in your doc seem more appropriate. It gets a little harder to read when there are huge comments like this, and this isn't necessary to understanding what the code does.

Reply:
By this I mean everything from "We could theoretically" to "leave the other callbacks as-is". The rest of the comment helps the viewer understand the code.